lib/src/internet_connection.dart

```diff
@@ -142,6 +146,20 @@ class InternetConnection {
     }
   }
 
+  /// Allows to change existing [_checkInterval].
+  ///
+  /// Resets the [_timerHandle] in the process.
+  void setIntervalAndResetTimer(Duration duration) {
+    _checkInterval = duration;
+    _timerHandle?.cancel();
+    _timerHandle = Timer(_checkInterval, _maybeEmitStatusUpdate);
+  }
+
+  /// Returns current [_checkInterval].
```

Comment:
Please update comment to `Returns the current duration between connection checks.`